### PR TITLE
T341589: Improve formatting of df_to_remarkup / add kwargs

### PR DESCRIPTION
**Pull request by andrewtavis-wmde**
## Changes

- `df_to_remarkup` now makes use of `pandas.DataFrame.to_markdown` to dramatically simplify the code
  - Adding `.replace(":", "-")` to the result is the only required change for Phabricator formatting
- This change also allows for kwargs for `pandas.DataFrame.to_markdown` and the Python package `tabulate` to be passed for formatting the results (including row numbers and formatting float columns)
- The doc string for the function has been changed to reflect the new usage and includes examples for the `index` and `floatfmt` kwargs
- Minor changes to indentation in `utils.py` in places where two spaces were being used instead of four

## Notes

- Please see [T341589](https://phabricator.wikimedia.org/T341589) for the discussion :)

### CODE REVIEW

**Comment by andrewtavis-wmde** [comment]
Hey @bearloga and @nettrom :wave: I've been discussing these changes with @nshahquinn who's on vacation in [T341589](https://phabricator.wikimedia.org/T341589). Was told that maybe one of you would be able to review, but also no rush on this!

Please let me know if there's anything I need to do on my end!

**Comment by andrewtavis-wmde** [comment]
Also, please let me know if a short write up of these changes in say `working-with-data` would be something that should also be done :)

File: wmfdata/utils.py [hunk 1]
```diff
@@ -52,10 +52,10 @@ def insert_code_toggle():
...
     <form action="javascript:code_toggle()">
         <input
-          id="code_toggle"
-          type="submit"
-          value="Hide code"
-          style="font-size: 1.4em"
+            id="code_toggle"
+            type="submit"
+            value="Hide code"
+            style="font-size: 1.4em"
```
Comment (xabriel) [inline wmfdata/utils.py]: Do we need these blank space changes or can we remove?

File: wmfdata/utils.py [hunk 2]
```diff
@@ -86,33 +86,21 @@ def mediawiki_dt(dt):
...
-    )
-
-    # Add a pipe to the start of every line, before adding the header separator
-    # so it doesn't get a double first pipe
-    remarkup_table = re.sub(r"^([^|])", r"| \1", psv_table, flags=re.MULTILINE)
-    # Make the first row a header
-    remarkup_table = remarkup_table.replace("\n", "\n" + header_sep + "\n", 1)
-
-    print(remarkup_table)
+    print(df.to_markdown(tablefmt="pipe", **kwargs).replace(":", "-"))
```
Comment (xabriel) [inline wmfdata/utils.py:102]: Nice!

File: wmfdata/utils.py [hunk 3]
```diff
@@ -140,7 +128,7 @@ def check_kerberos_auth():
         )
     elif klist != 0:
         raise OSError(
-          "There was an unknown issue checking your Kerberos credentials."
+            "There was an unknown issue checking your Kerberos credentials."
```
Comment (xabriel) [inline wmfdata/utils.py]: Ditto. Space changes look unrelated to contribution?

If we need to reformat, let's do it on a separate PR.

**Review by xabriel — COMMENTED** [review]
Thanks for the contribution @andrewtavis-wmde! I see from https://phabricator.wikimedia.org/T341589 that you have discussed this already.

I've left a couple comments above.

Also, can you please include one or two unit tests for this? The current test harness is very rudimentary, but you can see examples at https://github.com/wikimedia/wmfdata-python/tree/main/wmfdata_tests.

File: wmfdata/utils.py (hunk 1 above)
Comment (andrewtavis-wmde) [reply wmfdata/utils.py]: I would suggest that they be maintained as they're from improper indentation. All of the changes in indentation were from where two spaces were used instead of four :)

File: wmfdata/utils.py (hunk 3 above)
Comment (andrewtavis-wmde) [reply wmfdata/utils.py]: Ah ok, I didn't realize that they'd specifically need their own PR. Figured I'd just send them as indent rainbow was showing me that they were wrong. I guess these can be reverted then.

**Comment by andrewtavis-wmde** [comment]
Thanks for the feedback here, @xabriel! I'll go through and revert the changes to indentation and send along one or two unit tests as you suggested. Please let me know if an idea for a test jumps out at you and I'd be happy to write it up :) 

Also happy to go through and do full package reformat in a future PR! I saw that there are some other files that have places where indentation is off.

**Comment by andrewtavis-wmde** [comment]
Just made [T348999](https://phabricator.wikimedia.org/T348999) for the formatting work to make sure that we don't forget it. Figured it'd be quick enough, but then I realized that there are some link errors in readmes that could be checked for, and maybe others have ideas of things that should be done to improve code quality :)

**Comment by andrewtavis-wmde** [comment]
Hey @xabriel! :wave: c06b236 adds unit tests that check that we're getting the appropriate index and float formatting outputs given `kwarg` combinations passed to the `df_to_remarkup` :) I also made some minor changes to the docstring of `df_to_remarkup` so that we have an example like `floatfmt=(".0f", ".1f"))` that shows people how to format columns in different ways :blush:

Please let me know if there are other changes here that are needed! Also I'm hoping that we can excuse the removal of whitespace for certain lines in the testing file. I did revert the changes to indentation that were in the original commit and was actually trying to remove these whitespace edits just now, but then each time I removed another one the auto formatting I have set up would then reformat the last one I reverted... I'll go through and make the rest of the package consistent in [T348999](https://phabricator.wikimedia.org/T348999)!

**Comment by xabriel** [comment]
> https://github.com/wikimedia/wmfdata-python/commit/c06b2365a2160b8efe487afef957d3c4857e71aa adds unit tests that check that we're getting the appropriate index and float formatting outputs

great!

>  I'll go through and make the rest of the package consistent in [T348999](https://phabricator.wikimedia.org/T348999)!

Fair enough!

**Comment by xabriel** [comment]
Thanks for the contribution and for the test @andrewtavis-wmde ! Merging!

**Comment by andrewtavis-wmde** [comment]
Thanks for the review and the merge, @xabriel! I'll get back to you in [T348999](https://phabricator.wikimedia.org/T348999) about the next steps :)
